### DISCUSSION TITLE
Search Query for Videos accross a Hub site

### DISCUSSION THREAD

**Discussion by StfBauer** [microsoft-search/pnp-modern-search#3076] [Q&A] [answered]
Searching for videos across a hub site using the highlighted content web part works with the following query:

```kql
{searchterms} 
(DepartmentId:fe7f9ce4-ab95-4246-92a1-22833c558082 OR DepartmentId:{fe7f9ce4-ab95-4246-92a1-22833c558082}) (ContentTypeId:0x010100F3754F12A9B6490D9622A01FE9D8F012* OR  
(FileType:3gp OR FileType:asf OR FileType:avi OR FileType:mkv OR FileType:mod OR FileType:mov OR FileType:mp4 OR FileType:mpeg OR FileType:mpg OR FileType:mts OR FileType:ts OR FileType:vob OR FileType:wmv OR FileType:webm))
```

A search query across the whole tenant for video can also formulated like this

```kql
{searchterms} ContentType:video/*
```

This query will return any video store in the SharePoint tenant. However, the following query across a hub site will return nothing.

```kql
{searchTerms} 
(DepartmentId:fe7f9ce4-ab95-4246-92a1-22833c558082 OR DepartmentId:{fe7f9ce4-ab95-4246-92a1-22833c558082}) ContentType:video/*
```

Is there some logic behind file extensions vs search for all videos by mime type?

**Comment by kasperbolarsen** [comment]
Hi @StfBauer are you sure that {searchterms} ContentType:video/* is return anything? I just checked on my tenant, and the mp4 I check was showing 

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/cdb26454-0b6d-4f4f-b354-9888bb9c3979)

I am not sure why the query contains two departmentIds one with and one without the curly brackets?

**Comment by StfBauer** [comment, marked as answer]
Found the issue, and it works now. While you cannot query for Mime Types, the following query works.

```kql
{searchterms} 
DepartmentId:{Hub.Id}
(ContentTypeId:0x010100F3754F12A9B6490D9622A01FE9D8F012* OR  
(FileType:3gp OR FileType:asf OR FileType:avi OR FileType:mkv OR FileType:mod OR FileType:mov OR FileType:mp4 OR FileType:mpeg OR FileType:mpg OR FileType:mts OR FileType:ts OR FileType:vob OR FileType:wmv OR FileType:webm))
```

Stolen from highlighted content web part. Not optimal, but it is what it is.
